server/app/Main.hs

```diff
@@ -1,4 +1,46 @@
 import Lib
+import Options (Option (..), runOptionParser)
+import qualified Options.Applicative as Opt
+
+getConfig :: IO ServerConfig
+getConfig = runOptionParser $ \envOption ->
+  ServerConfig
+    <$> envOption
+      ( Option
+          { optHelp = "Path to the nix program database.",
+            optMetaVar = "PATH",
+            optLong = "program-db",
+            optShort = 'd',
+            optEnvKey = "NIX_PROGRAM_DB",
+            optReadM = ProgramDB <$> Opt.str,
+            optDefault = ProgramDB "/nix/var/nix/profiles/per-user/root/channels/nixos/programs.sqlite",
+            optShow = unProgramDB
+          }
+      )
+    <*> envOption
+      ( Option
+          { optHelp = "Path to the application database.",
+            optMetaVar = "PATH",
+            optLong = "app-db",
+            optShort = 'a',
+            optEnvKey = "APP_DB",
+            optReadM = ApplicationDB <$> Opt.str,
+            optDefault = ApplicationDB "appdb.sqlite",
+            optShow = unApplicationDB
+          }
+      )
+    <*> envOption
+      ( Option
+          { optHelp = "Port to listen on.",
+            optMetaVar = "INT",
+            optLong = "port",
+            optShort = 'p',
+            optEnvKey = "PORT",
```

Comment:
As for `PORT`, we have some convention and have less risk to conflict or get confused.

But I know that the convention is not so universal that it seems good to go `BINPLZ_PORT`.